### ISSUE TITLE
Kubelet migration to structured logs

### ISSUE THREAD

**Issue by ehashman** [kubernetes/kubernetes#98976]
For Structured Logging Migration: https://github.com/kubernetes/enhancements/issues/1602

[Beta graduation criteria:](https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging#beta)
- Structured logging interface design is verify by migrating one whole component (Kubelet) to structured logging. All other component migrations will be best effort.

### What would you like to be added:

Please migrate large chunks of code in each PR, at **minimum** one top level directory at a time (assuming your diff does not exceed ~40 lines). There are already ~40 top-level directories in the kubelet. If we have fewer PRs, it will be much easier for reviewers and approvers to merge your change.

The following query can be used to identify directories that still need structured logging migration:

```bash
git grep -L klog pkg/kubelet/ | sed 's|[^/]\+$||' | sort -u
git grep -L klog cmd/kubelet/ | sed 's|[^/]\+$||' | sort -u
```

You can check the project board to see if someone has already migrated a certain file: https://github.com/orgs/kubernetes/projects/53

#### Authors: What to include in your PR

- **Subject:** "Migrate `<directory>` to structured logging"
- **SIGs:** instrumentation, node
- **Priority:** important-soon
- **Kind:** cleanup
- **Release note:** You **must** include a release note for any log changes!

**HOWTO:** Read the [structured logging migration guide](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md).

_Only_ reference this issue for kubelet migrations. Don't mark it as "fixes" or other keywords it will close this tracker. All other components are best effort for 1.21 and should have priority/backlog.

Before you submit your PR, check to see if someone has already migrated that component: https://github.com/kubernetes/kubernetes/pulls?q=is%3Apr+is%3Aopen+structured+log+label%3Asig%2Fnode

#### Reviewers: What to look for

Read through the migration guide above. Use the PR query above to find PRs to review. Follow standard [node triage guidelines](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-node/triage.md).

### Why is this needed:

See above.

/sig instrumentation
/sig node
/cc @serathius 
/milestone v1.21

**Comment by neolit123** [comment]
> Release note: You must include a release note for any log changes!

this will make the release notes extremely noisy, given these PRs will be done per-package and not per-component.

**Comment by ehashman** [comment]
@neolit123 yeah it's somewhat unavoidable... I've reached out to a couple people on the release team, I think we'll end up squashing them down. There should be a limited number of ACTION REQUIRED notes (only for exit code changes in replacing fatal logger calls).

**Comment by erismaster** [comment]
Hello! I am from the Bug Triage team! Code freeze is starting March 9th, 2021 (about 1 week from now). We want to ensure that each PR has a chance to be merged before code freeze.
As the Issue is tagged for 1.21, is it still planned for this release?

**Comment by ehashman** [comment]
@erismaster it is.

Current status: the following files are still pending migration

- cmd/kubelet/app/plugins_providers.go NEEDS PR
- cmd/kubelet/app/server.go https://github.com/kubernetes/kubernetes/pull/98334
- cmd/kubelet/app/server_linux.go NEEDS PR
- pkg/kubelet/cadvisor/cadvisor_linux.go NEEDS PR
- pkg/kubelet/certificate/bootstrap/bootstrap.go https://github.com/kubernetes/kubernetes/pull/98993
- pkg/kubelet/certificate/transport.go https://github.com/kubernetes/kubernetes/pull/98993
- pkg/kubelet/cloudresource/cloud_request_manager.go https://github.com/kubernetes/kubernetes/pull/98999
- pkg/kubelet/cm/cgroup_manager_linux.go NEEDS PR
- pkg/kubelet/cm/container_manager_linux.go NEEDS PR
- pkg/kubelet/cm/container_manager_stub.go NEEDS PR
- pkg/kubelet/cm/cpumanager/cpu_assignment.go NEEDS PR
- pkg/kubelet/cm/cpumanager/cpu_manager.go NEEDS PR
- pkg/kubelet/cm/cpumanager/fake_cpu_manager.go NEEDS PR
- pkg/kubelet/cm/cpumanager/policy_none.go NEEDS PR
- pkg/kubelet/cm/cpumanager/policy_static.go NEEDS PR
- pkg/kubelet/cm/cpumanager/state/state_checkpoint.go https://github.com/kubernetes/kubernetes/pull/99563
- pkg/kubelet/cm/cpumanager/state/state_mem.go https://github.com/kubernetes/kubernetes/pull/99563
- pkg/kubelet/cm/cpumanager/topology/topology.go NEEDS PR
- pkg/kubelet/cm/cpuset/cpuset.go NEEDS PR
- pkg/kubelet/cm/devicemanager/device_plugin_stub.go NEEDS PR
- pkg/kubelet/cm/devicemanager/endpoint.go NEEDS PR
- pkg/kubelet/cm/devicemanager/manager.go NEEDS PR
- pkg/kubelet/cm/devicemanager/pod_devices.go NEEDS PR
- pkg/kubelet/cm/devicemanager/topology_hints.go https://github.com/kubernetes/kubernetes/pull/98632
- pkg/kubelet/cm/memorymanager/fake_memory_manager.go NEEDS PR
- pkg/kubelet/cm/memorymanager/memory_manager.go NEEDS PR
- pkg/kubelet/cm/memorymanager/policy_static.go NEEDS PR
- pkg/kubelet/cm/memorymanager/policy_static_test.go not sure why test is logging
- pkg/kubelet/cm/memorymanager/state/state_checkpoint.go NEEDS PR
- pkg/kubelet/cm/memorymanager/state/state_mem.go NEEDS PR
- pkg/kubelet/cm/node_container_manager_linux.go NEEDS PR
- pkg/kubelet/cm/pod_container_manager_linux.go NEEDS PR
- pkg/kubelet/cm/qos_container_manager_linux.go NEEDS PR
- pkg/kubelet/cm/topologymanager/fake_topology_manager.go NEEDS PR
- pkg/kubelet/cm/topologymanager/policy.go NEEDS PR
- pkg/kubelet/cm/topologymanager/scope_container.go NEEDS PR
- pkg/kubelet/cm/topologymanager/scope.go NEEDS PR
- pkg/kubelet/cm/topologymanager/scope_pod.go NEEDS PR
- pkg/kubelet/cm/topologymanager/topology_manager.go NEEDS PR
- pkg/kubelet/config/common.go NEEDS PR
- pkg/kubelet/config/config.go NEEDS PR
- pkg/kubelet/config/file.go NEEDS PR
- pkg/kubelet/config/file_linux.go NEEDS PR
- pkg/kubelet/config/http.go NEEDS PR
- pkg/kubelet/container/container_gc.go https://github.com/kubernetes/kubernetes/pull/98850
- pkg/kubelet/container/helpers.go https://github.com/kubernetes/kubernetes/pull/98850
- pkg/kubelet/container/runtime.go https://github.com/kubernetes/kubernetes/pull/98850
- pkg/kubelet/cri/remote/util/util_unix.go NEEDS PR
- pkg/kubelet/cri/streaming/portforward/httpstream.go https://github.com/kubernetes/kubernetes/pull/99006
- pkg/kubelet/cri/streaming/portforward/websocket.go https://github.com/kubernetes/kubernetes/pull/99006
- pkg/kubelet/cri/streaming/remotecommand/httpstream.go https://github.com/kubernetes/kubernetes/pull/99006
- pkg/kubelet/dockershim/network/cni/cni_windows.go NEEDS PR
- pkg/kubelet/eviction/eviction_manager.go https://github.com/kubernetes/kubernetes/pull/99029 https://github.com/kubernetes/kubernetes/pull/99032
- pkg/kubelet/eviction/helpers.go https://github.com/kubernetes/kubernetes/pull/99029 https://github.com/kubernetes/kubernetes/pull/99032
- pkg/kubelet/eviction/memory_threshold_notifier.go https://github.com/kubernetes/kubernetes/pull/99029 https://github.com/kubernetes/kubernetes/pull/99032
- pkg/kubelet/eviction/threshold_notifier_linux.go https://github.com/kubernetes/kubernetes/pull/99029 https://github.com/kubernetes/kubernetes/pull/99032
- pkg/kubelet/images/image_gc_manager.go https://github.com/kubernetes/kubernetes/pull/97520
- pkg/kubelet/images/image_manager.go NEEDS PR
- pkg/kubelet/kubeletconfig/configsync.go NEEDS PR
- pkg/kubelet/kubeletconfig/util/codec/codec.go NEEDS PR
- pkg/kubelet/kubeletconfig/util/log/log.go NEEDS PR
- pkg/kubelet/kubelet_getters.go NEEDS PR
- pkg/kubelet/kubelet.go NEEDS PR
- pkg/kubelet/kubelet_node_status.go https://github.com/kubernetes/kubernetes/pull/98154
- pkg/kubelet/kubelet_pods.go https://github.com/kubernetes/kubernetes/pull/98847
- pkg/kubelet/kubelet_resources.go NEEDS PR
- pkg/kubelet/kubelet_volumes.go https://github.com/kubernetes/kubernetes/pull/98850
- pkg/kubelet/kuberuntime/helpers.go NEEDS PR
- pkg/kubelet/kuberuntime/kuberuntime_container.go https://github.com/kubernetes/kubernetes/pull/96973
- pkg/kubelet/kuberuntime/kuberuntime_container_linux.go NEEDS PR
- pkg/kubelet/kuberuntime/kuberuntime_gc.go https://github.com/kubernetes/kubernetes/pull/97569
- pkg/kubelet/kuberuntime/kuberuntime_image.go NEEDS PR
- pkg/kubelet/kuberuntime/kuberuntime_manager.go NEEDS PR
- pkg/kubelet/kuberuntime/kuberuntime_sandbox.go NEEDS PR
- pkg/kubelet/kuberuntime/labels.go NEEDS PR
- pkg/kubelet/kuberuntime/logs/logs.go NEEDS PR
- pkg/kubelet/lifecycle/handlers.go https://github.com/kubernetes/kubernetes/pull/99479
- pkg/kubelet/lifecycle/predicate.go https://github.com/kubernetes/kubernetes/pull/99479
- pkg/kubelet/logs/container_log_manager.go NEEDS PR
- pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux.go https://github.com/kubernetes/kubernetes/pull/98847
- pkg/kubelet/nodeshutdown/systemd/inhibit_linux.go https://github.com/kubernetes/kubernetes/pull/98847
- pkg/kubelet/oom/oom_watcher_linux.go https://github.com/kubernetes/kubernetes/pull/99479
- pkg/kubelet/pluginmanager/cache/actual_state_of_world.go NEEDS PR
- pkg/kubelet/pluginmanager/cache/desired_state_of_world.go NEEDS PR
- pkg/kubelet/pluginmanager/operationexecutor/operation_generator.go NEEDS PR
- pkg/kubelet/pluginmanager/plugin_manager.go NEEDS PR
- pkg/kubelet/pluginmanager/pluginwatcher/example_handler.go NEEDS PR
- pkg/kubelet/pluginmanager/pluginwatcher/example_plugin.go NEEDS PR
- pkg/kubelet/pluginmanager/pluginwatcher/plugin_watcher.go NEEDS PR
- pkg/kubelet/pluginmanager/reconciler/reconciler.go NEEDS PR
- pkg/kubelet/pod_container_deletor.go https://github.com/kubernetes/kubernetes/pull/98850
- pkg/kubelet/pod_workers.go https://github.com/kubernetes/kubernetes/pull/98850
- pkg/kubelet/preemption/preemption.go NEEDS PR
- pkg/kubelet/prober/prober.go https://github.com/kubernetes/kubernetes/pull/97093
- pkg/kubelet/prober/prober_manager.go NEEDS PR
- pkg/kubelet/prober/prober_manager_test.go not sure why test is logging
- pkg/kubelet/prober/worker.go NEEDS PR
- pkg/kubelet/runonce.go https://github.com/kubernetes/kubernetes/pull/98850
- pkg/kubelet/server/server.go NEEDS PR
- pkg/kubelet/stats/cadvisor_stats_provider.go NEEDS PR
- pkg/kubelet/stats/cri_stats_provider.go NEEDS PR
- pkg/kubelet/stats/helper.go NEEDS PR
- pkg/kubelet/status/status_manager.go NEEDS PR
- pkg/kubelet/token/token_manager.go https://github.com/kubernetes/kubernetes/pull/99264
- pkg/kubelet/util/manager/watch_based_manager.go NEEDS PR
- pkg/kubelet/util/nodelease.go NEEDS PR
- pkg/kubelet/util/util_unix.go NEEDS PR
- pkg/kubelet/volume_host.go https://github.com/kubernetes/kubernetes/pull/98850
- pkg/kubelet/volumemanager/cache/actual_state_of_world.go https://github.com/kubernetes/kubernetes/pull/98850
- pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go https://github.com/kubernetes/kubernetes/pull/98850
- pkg/kubelet/volumemanager/reconciler/reconciler.go https://github.com/kubernetes/kubernetes/pull/98850
- pkg/kubelet/volumemanager/reconciler/reconciler_test.go not sure why we're calling klog in a test, this is easy to migrate/remove 
- pkg/kubelet/volumemanager/volume_manager.go https://github.com/kubernetes/kubernetes/pull/98850

**Comment by ehashman** [comment]
Here's the line by line breakdown of things that need PRs:

<details>

Lines | File
-- | --
3 | cmd/kubelet/app/plugins_providers.go
4 | cmd/kubelet/app/server_linux.go
2 | pkg/kubelet/cadvisor/cadvisor_linux.go
9 | pkg/kubelet/cm/cgroup_manager_linux.go
26 | pkg/kubelet/cm/container_manager_linux.go
1 | pkg/kubelet/cm/container_manager_stub.go
3 | pkg/kubelet/cm/cpumanager/cpu_assignment.go
17 | pkg/kubelet/cm/cpumanager/cpu_manager.go
8 | pkg/kubelet/cm/cpumanager/fake_cpu_manager.go
1 | pkg/kubelet/cm/cpumanager/policy_none.go
15 | pkg/kubelet/cm/cpumanager/policy_static.go
1 | pkg/kubelet/cm/cpumanager/topology/topology.go
1 | pkg/kubelet/cm/cpuset/cpuset.go
9 | pkg/kubelet/cm/devicemanager/device_plugin_stub.go
4 | pkg/kubelet/cm/devicemanager/endpoint.go
36 | pkg/kubelet/cm/devicemanager/manager.go
16 | pkg/kubelet/cm/devicemanager/pod_devices.go
8 | pkg/kubelet/cm/memorymanager/fake_memory_manager.go
10 | pkg/kubelet/cm/memorymanager/memory_manager.go
19 | pkg/kubelet/cm/memorymanager/policy_static.go
10 | pkg/kubelet/cm/memorymanager/policy_static_test.go
5 | pkg/kubelet/cm/node_container_manager_linux.go
8 | pkg/kubelet/cm/pod_container_manager_linux.go
8 | pkg/kubelet/cm/qos_container_manager_linux.go
6 | pkg/kubelet/cm/topologymanager/fake_topology_manager.go
3 | pkg/kubelet/cm/topologymanager/policy.go
4 | pkg/kubelet/cm/topologymanager/scope_container.go
1 | pkg/kubelet/cm/topologymanager/scope.go
4 | pkg/kubelet/cm/topologymanager/scope_pod.go
2 | pkg/kubelet/cm/topologymanager/topology_manager.go
4 | pkg/kubelet/config/common.go
9 | pkg/kubelet/config/config.go
9 | pkg/kubelet/config/file.go
2 | pkg/kubelet/config/file_linux.go
5 | pkg/kubelet/config/http.go
1 | pkg/kubelet/cri/remote/util/util_unix.go
1 | pkg/kubelet/images/image_manager.go
1 | pkg/kubelet/kubeletconfig/configsync.go
1 | pkg/kubelet/kubeletconfig/util/codec/codec.go
2 | pkg/kubelet/kubeletconfig/util/log/log.go
3 | pkg/kubelet/kubelet_getters.go
64 | pkg/kubelet/kubelet.go
1 | pkg/kubelet/kubelet_resources.go
1 | pkg/kubelet/kuberuntime/helpers.go
2 | pkg/kubelet/kuberuntime/kuberuntime_container_linux.go
6 | pkg/kubelet/kuberuntime/kuberuntime_image.go
47 | pkg/kubelet/kuberuntime/kuberuntime_manager.go
9 | pkg/kubelet/kuberuntime/kuberuntime_sandbox.go
11 | pkg/kubelet/kuberuntime/labels.go
9 | pkg/kubelet/kuberuntime/logs/logs.go
7 | pkg/kubelet/logs/container_log_manager.go
1 | pkg/kubelet/pluginmanager/cache/actual_state_of_world.go
1 | pkg/kubelet/pluginmanager/cache/desired_state_of_world.go
2 | pkg/kubelet/pluginmanager/operationexecutor/operation_generator.go
3 | pkg/kubelet/pluginmanager/plugin_manager.go
1 | pkg/kubelet/pluginmanager/pluginwatcher/example_handler.go
7 | pkg/kubelet/pluginmanager/pluginwatcher/example_plugin.go
14 | pkg/kubelet/pluginmanager/pluginwatcher/plugin_watcher.go
7 | pkg/kubelet/pluginmanager/reconciler/reconciler.go
3 | pkg/kubelet/preemption/preemption.go
3 | pkg/kubelet/prober/prober_manager.go
3 | pkg/kubelet/prober/prober_manager_test.go
6 | pkg/kubelet/prober/worker.go
5 | pkg/kubelet/server/server.go
4 | pkg/kubelet/stats/cadvisor_stats_provider.go
10 | pkg/kubelet/stats/cri_stats_provider.go
1 | pkg/kubelet/stats/helper.go
34 | pkg/kubelet/status/status_manager.go
1 | pkg/kubelet/util/manager/watch_based_manager.go
1 | pkg/kubelet/util/nodelease.go
1 | pkg/kubelet/util/util_unix.go

</details>



I suggest we split it up as follows:

- cmd/kubelet (S)
- pkg/kubelet/cadvisor/cadvisor_linux.go + pkg/kubelet/cri/remote/util/util_unix.go + pkg/kubelet/images/image_manager.go (XS)
- pkg/kubelet/config (M)
- pkg/kubelet/kubeletconfig (XS)
- pkg/kubelet/kubelet_getters.go + pkg/kubelet/kubelet_resources.go (XS)
- pkg/kubelet/kubelet.go (L)
- pkg/kubelet/kuberuntime/kuberuntime_manager.go (L)
- pkg/kubelet/kuberuntime except for runtime_manager.go (M)
- pkg/kubelet/logs/container_log_manager.go (S)
- pkg/kubelet/pluginmanager (M)
- pkg/kubelet/preemption/preemption.go (XS)
- pkg/kubelet/prober (S)
- pkg/kubelet/server/server.go (S)
- pkg/kubelet/stats (M)
- pkg/kubelet/status/status_manager.go (M)
- pkg/kubelet/util/manager/watch_based_manager.go + pkg/kubelet/util/nodelease.go + pkg/kubelet/util/util_unix.go (XS)

**Comment by shiyajuan123** [comment]
/assign
working on these:
pkg/kubelet/cm/container_manager_linux.go
pkg/kubelet/cm/container_manager_stub.go

**Comment by adisky** [comment]
Everyone working on the issue please comment here on what packages you are working to avoid duplicate efforts and it will be easier for reviewers :).

**Comment by qingwave** [comment]
/assign
working on these:
pkg/kubelet/preemption/preemption.go

**Comment by navidshaikh** [comment]
/assign
working on: 
#99823 pkg/kubelet/util/manager/watch_based_manager.go + pkg/kubelet/util/nodelease.go + pkg/kubelet/util/util_unix.go (XS)
pkg/kubelet/status/status_manager.go (M)

**Comment by yuzhiquan** [comment]
If [err == nil](https://github.com/kubernetes/klog/blob/110d443dcd6b29c2941cabd9ab9d211063658a21/klog.go#L809)  will result in 
```
klog.Errors(nil, "xxxxxxxxx")
>>>
I0102 15:04:05.067890    1234 prober_manager.go:956] "xxxxxxxxxxxxxxxxxxx"
```
This would change log level.
May be we should use `InfoS` instead of `Erros`.
@ehashman 
/cc @serathius

**Comment by serathius** [comment]
@yuzhiquan Can you file an issue in klog repo?

**Comment by adisky** [comment]
Working on `pkg/kubelet/server/server.go (S)`

**Comment by serathius** [comment]
Please cc me in PRs, so we can coverage of reviewers from different timezones.

**Comment by sladyn98** [comment]
Depending on the above comments I have updated the list. Let me know if I missed something

Lines | File
-- | --
3 | cmd/kubelet/app/plugins_providers.go
4 | cmd/kubelet/app/server_linux.go
2 | pkg/kubelet/cadvisor/cadvisor_linux.go
9 | pkg/kubelet/cm/cgroup_manager_linux.go
3 | pkg/kubelet/cm/cpumanager/cpu_assignment.go
17 | pkg/kubelet/cm/cpumanager/cpu_manager.go
8 | pkg/kubelet/cm/cpumanager/fake_cpu_manager.go
1 | pkg/kubelet/cm/cpumanager/policy_none.go
15 | pkg/kubelet/cm/cpumanager/policy_static.go
1 | pkg/kubelet/cm/cpumanager/topology/topology.go
1 | pkg/kubelet/cm/cpuset/cpuset.go
9 | pkg/kubelet/cm/devicemanager/device_plugin_stub.go
4 | pkg/kubelet/cm/devicemanager/endpoint.go
36 | pkg/kubelet/cm/devicemanager/manager.go
16 | pkg/kubelet/cm/devicemanager/pod_devices.go
8 | pkg/kubelet/cm/memorymanager/fake_memory_manager.go
10 | pkg/kubelet/cm/memorymanager/memory_manager.go
19 | pkg/kubelet/cm/memorymanager/policy_static.go
10 | pkg/kubelet/cm/memorymanager/policy_static_test.go
5 | pkg/kubelet/cm/node_container_manager_linux.go
8 | pkg/kubelet/cm/pod_container_manager_linux.go
8 | pkg/kubelet/cm/qos_container_manager_linux.go
6 | pkg/kubelet/cm/topologymanager/fake_topology_manager.go
3 | pkg/kubelet/cm/topologymanager/policy.go
4 | pkg/kubelet/cm/topologymanager/scope_container.go
1 | pkg/kubelet/cm/topologymanager/scope.go
4 | pkg/kubelet/cm/topologymanager/scope_pod.go
2 | pkg/kubelet/cm/topologymanager/topology_manager.go
4 | pkg/kubelet/config/common.go
9 | pkg/kubelet/config/config.go
9 | pkg/kubelet/config/file.go
2 | pkg/kubelet/config/file_linux.go
5 | pkg/kubelet/config/http.go
1 | pkg/kubelet/cri/remote/util/util_unix.go
1 | pkg/kubelet/images/image_manager.go
1 | pkg/kubelet/kubeletconfig/configsync.go
1 | pkg/kubelet/kubeletconfig/util/codec/codec.go
2 | pkg/kubelet/kubeletconfig/util/log/log.go
3 | pkg/kubelet/kubelet_getters.go
64 | pkg/kubelet/kubelet.go
1 | pkg/kubelet/kubelet_resources.go
1 | pkg/kubelet/kuberuntime/helpers.go
2 | pkg/kubelet/kuberuntime/kuberuntime_container_linux.go
6 | pkg/kubelet/kuberuntime/kuberuntime_image.go
9 | pkg/kubelet/kuberuntime/kuberuntime_sandbox.go
11 | pkg/kubelet/kuberuntime/labels.go
9 | pkg/kubelet/kuberuntime/logs/logs.go
7 | pkg/kubelet/logs/container_log_manager.go
1 | pkg/kubelet/pluginmanager/cache/actual_state_of_world.go
1 | pkg/kubelet/pluginmanager/cache/desired_state_of_world.go
2 | pkg/kubelet/pluginmanager/operationexecutor/operation_generator.go
3 | pkg/kubelet/pluginmanager/plugin_manager.go
1 | pkg/kubelet/pluginmanager/pluginwatcher/example_handler.go
7 | pkg/kubelet/pluginmanager/pluginwatcher/example_plugin.go
14 | pkg/kubelet/pluginmanager/pluginwatcher/plugin_watcher.go
7 | pkg/kubelet/pluginmanager/reconciler/reconciler.go
3 | pkg/kubelet/prober/prober_manager.go
3 | pkg/kubelet/prober/prober_manager_test.go
6 | pkg/kubelet/prober/worker.go
4 | pkg/kubelet/stats/cadvisor_stats_provider.go
10 | pkg/kubelet/stats/cri_stats_provider.go
1 | pkg/kubelet/stats/helper.go

**Comment by navidshaikh** [comment]
Working on migrating `pkg/kubelet/kubelet.go`

**Comment by hexxdump** [comment]
Working on migrating pkg/kubelet/winstats

**Comment by yuzhiquan** [comment]
> @yuzhiquan Can you file an issue in klog repo?

@serathius so we  should accept ErrorS can take err =nil as an parameter?

**Comment by ehashman** [comment]
> @serathius so we should accept ErrorS can take err =nil as an parameter?

Yup, that's part of the migration guide. klog should not be converting `ErrorS(nil, "msg")` into info level logs.

**Comment by git001** [comment]
looks like the plan is to use something similar like Key-Value-Pair structure.
If this is the case then I think there should be the primary message also a Key-Value-Pair.

Cite from https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#verify-log-output

```
I0528 19:15:22.737588   47512 logtest.go:55] "Received HTTP request" verb="GET" URI="/metrics" latency="1s" resp=200 userAgent="Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0. 2272.118 Safari/537.36." srcIP="127.0.0.1"
```

Should it be the "Received HTTP request" written as **message="Received HTTP request"**
```
I0528 19:15:22.737588   47512 logtest.go:55] message="Received HTTP request" verb="GET" URI="/metrics" latency="1s" resp=200 userAgent="Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0. 2272.118 Safari/537.36." srcIP="127.0.0.1"
```

**Comment by serathius** [comment]
@git001 Thanks for feedback, format for `text` log was based on existing patterns used in K8s code base, were contributors were using `key=value` suffix even when using string formatting in `Infof`. We decided that we will skip naming the key for `message` as it would not provide any benefits and would diverge us more from existing pattern.

Main purpose for `text` format was to provide sensible backward compatibility for users of `glog` format, we may decide to put some improvements, but our main focus would be on JSON format, which should be much easier to parse and consume. 

Let's not continue this discussion on issue related to migrating Kubelet code base and move it to more convenient place. Feel free to create a new issue with any improvements you would want to propose so we can involve more people and discuss it properly.

**Comment by ehashman** [comment]
We still have a bunch of sections that are not yet migrated. The largest appears to be pkg/kubelet/cm. I am updating this list so it is accurate: https://github.com/kubernetes/kubernetes/issues/98976#issuecomment-790927982

**Comment by krzysiekg** [comment]
I'm working on pkg/kubelet/kuberuntime

**Comment by ehashman** [comment]
/help

Please ensure you look at https://github.com/kubernetes/kubernetes/issues/98976#issuecomment-790927982 before you migrate files, read the instruction, and link back to this issue!

**Comment by knabben** [comment]
Working on `pkg/kubelet/cm/topologymanager`

**Comment by knabben** [comment]
Working on `pkg/kubelet/cm/memorymanager/`.

less `pkg/kubelet/cm/memorymanager/policy_static_test.go` that's already migrated.

**Comment by knabben** [comment]
Working on `pkg/kubelet/cm/devicemanager/`
less `pkg/kubelet/cm/devicemanager/topology_hints.go`

**Comment by AfrouzMashayekhi** [comment]
Working on `cmd/kubelet/app/plugins_providers.go` + `cmd/kubelet/app/server_linux.go` + `pkg/kubelet/cadvisor/cadvisor_linux.go` + `pkg/kubelet/cri/remote/util/util_unix.go` + `pkg/kubelet/images/image_manager.go`

**Comment by ehashman** [comment]
There are 3 more areas it looks like we have left to migrate, updated https://github.com/kubernetes/kubernetes/issues/98976#issuecomment-790956023 and the PR list

**Comment by AfrouzMashayekhi** [comment]
Working on `pkg/kubelet/cm/cpumanager`

**Comment by pacoxu** [comment]
I think #99996 is working on pkg/kubelet/dockershim/network/cni/cni_windows.go

**Comment by lala123912** [comment]
work on pkg/kubelet/cm/cgroup_manager_linux.go

**Comment by ehashman** [comment]
We have one remaining area to migrate:

- pkg/kubelet/cm/cpumanager

**Comment by lala123912** [comment]
work on pkg/kubelet/cm/cpumanager

**Comment by pacoxu** [comment]
pkg/kubelet/cm/cpumanager/topology/topology.go NEEDS PR
pkg/kubelet/cm/cpumanager/cpu_assignment.go NEEDS PR
pkg/kubelet/cm/cpumanager/policy_none.go NEEDS PR
#100163

pkg/kubelet/cm/cpumanager/cpu_manager.go NEEDS PR
pkg/kubelet/cm/cpumanager/fake_cpu_manager.go NEEDS PR
pkg/kubelet/cm/cpumanager/policy_static.go NEEDS PR

I will take it .

**Comment by serathius** [comment]
Since we added static analysis all directories migrated we should add migrated directories to https://github.com/kubernetes/kubernetes/blob/master/hack/.structured_logging 

Doing it in every PR will delay merging, as I expect that /hack/ has different approvers then kubelet. I will try to prepare separate PR for that.

**Comment by ehashman** [comment]
We are just waiting on approvers now! :tada: 
/remove-help
